contracts/contracts/coordination/TACoChildApplication.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./ITACoRootToChild.sol";
+import "../../threshold/ITACoChildApplication.sol";
+import "./ITACoChildToRoot.sol";
+
+/**
+ * @title TACoChildApplication
+ * @notice TACoChildApplication
+ */
+contract TACoChildApplication is AccessControl, ITACoRootToChild, ITACoChildApplication {
+    bytes32 public constant UPDATE_ROLE = keccak256("UPDATE_ROLE");
+    bytes32 public constant DEPLOYER_ROLE = keccak256("DEPLOYER_ROLE");
+
+    struct StakingProviderInfo {
+        address operator;
+        bool operatorConfirmed;
+        uint96 authorized;
+        // TODO: what about undelegations etc?
+    }
+
+    ITACoChildToRoot public immutable rootApplication;
+    address public coordinator;
+
+    mapping(address => StakingProviderInfo) public stakingProviderInfo;
+    mapping(address => address) public stakingProviderFromOperator;
+
+    constructor(ITACoChildToRoot _rootApplication, address[] memory updaters) {
+        require(
+            address(_rootApplication) != address(0),
+            "Address for root application must be specified"
+        );
+        rootApplication = _rootApplication;
+
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        for (uint256 i = 0; i < updaters.length; i++) {
+            _grantRole(UPDATE_ROLE, updaters[i]);
+        }
+    }
+
+    function setCoordinator(address _coordinator) external onlyRole(DEPLOYER_ROLE) {
+        require(coordinator == address(0), "Coordinator already set");
+        require(_coordinator != address(0), "Coordinator must be specified");
+        // require(_coordinator.numberOfRituals() >= 0, "Invalid coordinator");
```

Comment:
I see this `require` is commented out, but I wanted to ask what would motivate this check. Does `Coordinator` need to be in a certain state (has at least one completed ritual) to be considered a valid `Coordinator` in the app? Why? Is there some data/state dependency? between the two?

Reply:
similar checks in other places work as proof that provider contract has proper abi (or at least one method)

Reply:
Uncommented and updated that check